assets/src/amp-story-editor-blocks.js

```diff
@@ -20,6 +20,9 @@ const { getSelectedBlockClientId, getBlockOrder, getBlocksByClientId } = select(
 domReady( () => {
 	setDefaultBlockName( 'amp/amp-story-page' );
 
+	// Remove all blocks that aren't whitelisted.
+	getBlockTypes().filter( ( { name } ) => ! ALLOWED_BLOCKS.includes( name ) && 'amp/amp-story-page' ).map( ( { name } ) => unregisterBlockType( name ) );
```

Comment:
Minor point, but I think it would be better to not use `filter` as a control structure. In other words, I think it would be better to do:

```js
for ( const block of getBlockTypes().filter( ( { name } ) => ! ALLOWED_BLOCKS.includes( name ) && 'amp/amp-story-page' ) )  {
    unregisterBlockType( block.name );
}
```

But take a look at the condition here: `! ALLOWED_BLOCKS.includes( name ) && 'amp/amp-story-page'`. Does it make sense? Does does it intend to say rather `! ALLOWED_BLOCKS.includes( name ) && name !== 'amp/amp-story-page'`?

Reply:
That condition is indeed flawed. `&& 'amp/amp-story-page'` does nothing really. It would need to be `name !== 'amp/amp-story-page'`.

However, I later changed the code so that `ALLOWED_BLOCKS` includes `amp/amp-story-page` as well, which means I can remove that check without a replacement.

As for `map` vs. `for` I'll try to reply at #1904